scripts/logbook.py

```diff
@@ -232,26 +242,43 @@ def main(lims, pid, epp_logger):
                 log.append(instrument_name)
                 if record["details"][instrument_number] is not '':
                     for item in record["details"][instrument_number]:
-                        udf_detail.append(pro.udf[item[4:]])
-                    log.append(','.join(udf_detail))
+                        try:
+                            udf_content = pro.udf[item[4:]]
+                            udf_detail.append(item[4:]+":"+pro.udf[item[4:]])
+                        except KeyError, e:
+                            continue
+                    if udf_detail == []:
+                        log.append("-")
+                    else:
+                        log.append(','.join(udf_detail))
                     write_record(log,record["dest_file"][instrument_number])
                 else:
-                    log.append("")
+                    log.append("-")
                     write_record(log,record["dest_file"][instrument_number])
 
         elif instrument[0:3] == "udf":
-            if pro.udf[instrument[4:]] is not '':
+            try:
                 instrument_name = pro.udf[instrument[4:]]
-                log.append(instrument_name)
-                if record["details"][instrument_number] is not '':
-                    for item in record["details"][instrument_number]:
-                        udf_detail.append(pro.udf[item[4:]])
-                    log.append(','.join(udf_detail))
-                    write_record(log,record["dest_file"][instrument_number])
+                if pro.udf[instrument[4:]] is not '':
+                    log.append(instrument_name)
+                    if record["details"][instrument_number] is not '':
+                        for item in record["details"][instrument_number]:
```

Comment:
This `for` loop is exactly same as above, why not put it in a function to reduce redundant code